### PR TITLE
mec15xx: pinctrl for both mec15xx and mec17xx

### PR DESCRIPTION
**Pull request by jvasanth1**
use custom private header file pinctrl_mchp_xec_pvt.h to compile pinctrl driver for both mec15xx and mec17xx.
Also add mec15xx pinctrl dtsi

This is initial step for deprecating pinmux for mec15xx

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

### CODE REVIEW

File: drivers/pinctrl/pinctrl_mchp_xec_pvt.h [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ * Copyright (c) 2019 Intel Corporation
+ * Copyright (c) 2022 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MCHP_XEC_PVT_H
+#define __MCHP_XEC_PVT_H
+
+#ifndef _ASMLANGUAGE
```
Comment (gmarull) [inline drivers/pinctrl/pinctrl_mchp_xec_pvt.h:11]: why is this needed?

File: drivers/pinctrl/pinctrl_mchp_xec_pvt.h [hunk 2]
```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ * Copyright (c) 2022 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MCHP_XEC_PVT_H
```
Comment (gmarull) [inline drivers/pinctrl/pinctrl_mchp_xec_pvt.h:8]: note: no content in this header is shared, so please just put everything in the source file using this content (also, this header is not self-contained)

**Review by gmarull — CHANGES_REQUESTED** [review]
lgtm, minor comments

File: drivers/pinctrl/pinctrl_mchp_xec_pvt.h (hunk 2 above)
Comment (jvasanth1) [reply drivers/pinctrl/pinctrl_mchp_xec_pvt.h:8]: ok, will move header file contents to source file

File: drivers/pinctrl/pinctrl_mchp_xec_pvt.h (hunk 1 above)
Comment (jvasanth1) [reply drivers/pinctrl/pinctrl_mchp_xec_pvt.h]: updated

**Comment by hunterhu3000** [comment]
@jvasanth1 How to test this patch?

**Comment by jvasanth1** [comment]
> @jvasanth1 How to test this patch?

For MEC15xx pinctrl, compiling the pinctrl driver for MEC15xx is first step, subsequently we will be updating MEC15xx drivers (one by one) to use pinctrl instead of pinmux.

For MEC17xx, tested the changes by verifying an existing driver using pinctrl works as expected.

**Review by gmarull — APPROVED** [review]

**Comment by gmarull** [comment]
@jvasanth1 please rebase, GH UI says "This branch has conflicts that must be resolved"

File: drivers/pinctrl/Kconfig.xec [hunk 3]
```diff
@@ -5,7 +5,7 @@ DT_COMPAT_ST_PINCTRL_XEC := microchip,xec-pinctrl
 
 config PINCTRL_MCHP_XEC
 	bool "Pin controller driver for MCHP XEC MCUs"
-	depends on SOC_SERIES_MEC172X
+	depends on SOC_FAMILY_MEC
 	default $(dt_compat_enabled,$(DT_COMPAT_ST_PINCTRL_XEC))
```
Comment (gmarull) [inline drivers/pinctrl/Kconfig.xec:9]:
```suggestion
	default y
	depends on DT_HAS_MICROCHIP_XEC_PINCTRL_ENABLED
```

(and delete L4)

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/pinctrl/Kconfig.xec (hunk 3 above)
Comment (galak) [reply drivers/pinctrl/Kconfig.xec:9]: I think a rebase is needed since upstream already does this.

Comment (jvasanth1) [reply drivers/pinctrl/Kconfig.xec]: after the rebase, I am getting compilation errors, not sure why, working on it.
The MEC15xx build is compiling xec_gpio_v2 driver meant for mec17xx

Comment (jvasanth1) [reply drivers/pinctrl/Kconfig.xec]: the gpio_v2 driver was getting pulled due to the dts; fixed it in `mec1501hsz.dtsi`

**Comment by jvasanth1** [comment]
@gmarull please help to reapprove

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
